### PR TITLE
OSASINFRA-3278: add tests when etcd runs on additional local disk

### PR DESCRIPTION
**Pull request by EmilienM**
If both conditions are met:
* Machine API operator is deployed and managing the Machines
* a CPMS exists with an additional block device named `etcd`

Then we will check that:
* the Control Plane flavor in Nova has enough Ephemeral disk for storing etcd data
* all the Control Plane Machines have an ephemeral disk for etcd, of type `Local` and with the minimum size required

### CODE REVIEW

**Comment by EmilienM** [comment]
```
: [sig-installer][Suite:openshift/openstack] The OpenStack platform checks resources when etcd is on an ephemeral block device

Reason: skip [github.com/openshift/openstack-test/test/extended/openstack/local-bd-etcd.go:151]: CPMS does not have additional block devices Ginkgo exit error 3: exit with code 3
```

This is now skipped as expected.

I'm going to add a new CI job to test it, I'm not sure what we want first : the test or the job?

File: test/extended/openstack/local-bd-etcd.go
Comment (pierreprinetti) [inline test/extended/openstack/local-bd-etcd.go:24]: Please keep variables local to their context:

```suggestion
var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenStack platform", func() {
	defer g.GinkgoRecover()

	const minEtcdDiskSizeGiB = 7
	const etcdBlockDeviceName = "etcd"
```

File: test/extended/openstack/local-bd-etcd.go
Comment (pierreprinetti) [inline test/extended/openstack/local-bd-etcd.go:52]: This will render as:
> "[sig-installer][Suite:openshift/openstack] The OpenStack platform checks resources when etcd is on an ephemeral block device

which makes little sense. Can you please rephrase in terms of what you expect to be in the cluster?

File: test/extended/openstack/local-bd-etcd.go
Comment (pierreprinetti) [inline test/extended/openstack/local-bd-etcd.go:105]: If this function is only used in this test, consider moving it in the context of the test itself:

```Go
var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenStack platform", func() {
	defer g.GinkgoRecover()
	checkEtcdDisk := func(machine obj.Map) error { 
```

File: test/extended/openstack/local-bd-etcd.go
Comment (pierreprinetti) [inline test/extended/openstack/local-bd-etcd.go:85]: FYI I am experimenting a convenient API for getting machines (including by role, like in this case) here https://github.com/openshift/openstack-test/pull/127

File: test/extended/openstack/local-bd-etcd.go
Comment (EmilienM) [reply test/extended/openstack/local-bd-etcd.go:24]: No because they are used in functions too at the end of the file and if I move them, it doesn't compile.

File: test/extended/openstack/local-bd-etcd.go
Comment (EmilienM) [reply test/extended/openstack/local-bd-etcd.go:85]: Really cool and useful, I'll rebase on top of your PR if that merges before or I'll come back to it if mine merges before.

File: test/extended/openstack/local-bd-etcd.go
Comment (EmilienM) [reply test/extended/openstack/local-bd-etcd.go:105]: FYI I moved `getControlPlaneMachines` too.

File: test/extended/openstack/local-bd-etcd.go
Comment (pierreprinetti) [inline test/extended/openstack/local-bd-etcd.go:66]: I am not sure this is correct. At a minimum, shouldn't you look into a `labels` property?

Anyway, you can start using https://github.com/openshift/openstack-test/pull/127 as soon as we merge it (and if I got it wrong, then we will collectively notice & fix!)

Comment (EmilienM) [reply test/extended/openstack/local-bd-etcd.go:66]: yeah I'll refactor on your PR, yours will merge today I think. I'm waiting for CI and I'll approve.
Thanks for your work!

**Comment by pierreprinetti** [comment]
/lgtm
Under the (tested) assumption that this doesn’t bring regressions to the existing jobs, and as a base for the new job.

**Review by pierreprinetti — APPROVED** [review]

**Comment by EmilienM** [comment]
NOTE: this doesn't work. I tested locally and the test is skipped because it can't find the CPMS data. I'll fix it in another PR once https://github.com/openshift/release/pull/44991 has landed.

**Comment by pierreprinetti** [comment]
@itzikb-redhat has some code that pulls the CPMS here: https://github.com/openshift/openstack-test/pull/124

File: test/extended/openstack/local-bd-etcd.go
Comment (mandre) [reply test/extended/openstack/local-bd-etcd.go]: #127 has merged, should we add the changes Pierre asked for in https://github.com/openshift/openstack-test/pull/128 ?

Comment (EmilienM) [reply test/extended/openstack/local-bd-etcd.go]: that's what I did in the other PR? I think your comment it outdated

**Pull request merged**
